### ISSUE TITLE
Dedicated scheduler

### ISSUE THREAD

**Issue by bstrdsmkr** [dask/dask-jobqueue#102]
Hi folks,

I'm working on making it easier for our users to make use of our cluster (Stratus).  Stratus is a traditional cluster with a shared login node and several compute nodes. Aside from that, we have a node setup hosting Jupyterhub/lab which is configured as a submit host (`qsub test.pbs` can be ran in the Jupyter terminal to submit a job).

Our original intended work flow was that users could develop their algorithm in Jupyter using a subset of our data, then stage the full dataset to LUSTRE and submit their `nbconvert`ed script to the cluster all without leaving their browser.

I'd like to take that a step further and keep an active Dask scheduler running all the time so that our users can just connect a client through their notebook and submit tasks, but we can also leverage that same scheduler for operational and infrastructure tasks (we do a lot of processing to produce datasets).

This feels like I could just create a systemd service on the Jupyter host which maintains the scheduler, then users would skip creating the scheduler in their notebook and connect directly to the existing scheduler to run their tasks.

What am I missing?

**Comment by mrocklin** [comment]
That's definitely possible, and is a workflow used by some groups.  We've found that it's a bit more common for people to want to start up their own scheduler though.  There are some challenges to using a global scheduler:

1.   Software versions need to match, and users tend to want to use non-uniform versions of various libraries
2.   Multi-tenancy, some users may hog resources, so it's nice to engage the resource manager when making separate clusters

The workflow that we've seen become common is to use this library, dask-jobqueue, to create your own dask cluster.

```python
from dask_jobqueue import PBSCluster
cluster = PBSCluster()
cluster.scale(100)  # ask for 100 workers

from dask.distributed import client
client = Client(cluster)
```

This does force them to create their own scheduler (within the Cluster object) but this doesn't seem to be an onerous task or tripping point for most users.

You could still use your central dask scheduler for infrastructural tasks, you just become a peer to all the rest of the users.

But as I said, both models have been used by different groups.

**Comment by bstrdsmkr** [comment]
Ahh, I think I've made some bad assumptions about the ecosystem. So its ok to have more than one scheduler running for the same HPC cluster?

Software versions will be less of a problem for us specifically as we'll have fairly good control of what is available and make liberal use of conda envs and cross mounting.

Multi-tenancy is something we're still feeling out with respect to Dask. If I got my way, it'd be all Dask all the time, but as is some users are still going to want to use traditional PBS/qsub.

I think our ideal scenario would be setting the Dask scheduler adaptive and letting it scale down to 0 when unused, or up to the whole ~4k cores if available.

**Comment by mrocklin** [comment]
> Ahh, I think I've made some bad assumptions about the ecosystem. So its ok to have more than one scheduler running for the same HPC cluster?

Definitely.  It's common for each user to run their own dask scheduler.  These can be temporary and ephemeral.  

> Software versions will be less of a problem for us specifically as we'll have fairly good control of what is available and make liberal use of conda envs and cross mounting.

Right, but your users might also want to do this.  Different users might want to use scikit-learn version 0.19 and 0.20.  A single dask cluster must have uniform versioning, so these two users can't use the same.  It typically makes sense for each user to create their own ephemeral cluster.  This should only take a few seconds if the queue allows small jobs through.

> Multi-tenancy is something we're still feeling out with respect to Dask. If I got my way, it'd be all Dask all the time, but as is some users are still going to want to use traditional PBS/qsub.

To be clear, these aren't mutually exclusive.  Typically people launch dask workers through PBS/qsub.  

> I think our ideal scenario would be setting the Dask scheduler adaptive and letting it scale down to 0 when unused, or up to the whole ~4k cores if available.

Yup, users can also do this on their own, using PBS/qsub to manage resources.

**Comment by guillaumeeb** [comment]
@bstrdsmkr do you still have questions on this topic? Has @mrocklin answered to your need?
